newrelic-alerts/NewRelic_alerts.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+#
+# This file is managed by Dogsbody Technology Ltd.
+#   https://www.dogsbody.com/
+#
+# Description: A script to automate the process of setting up New Relic alerts
+#
+# Notes:
+# 	- Enter API key and json file name outside of script
+#	- If no json file is given, default is used
+#	- Commit to https://github.com/dogsbodytech/workstation-tools/
+#	- Write wiki documentation
+
+import requests
+import json
+import itertools
+from requests.structures import CaseInsensitiveDict
+
+# Defines the required New Relic API links
+CHANNEL_URL = "https://api.newrelic.com/v2/alerts_channels.json"
+POLICY_URL = "https://api.newrelic.com/v2/alerts_policies.json"
+CONDITIONS_URL = "https://infra-api.newrelic.com/v2/alerts/conditions"
+
+# Headers must be case insensitive for the request commands below
+headers = CaseInsensitiveDict()
+headers["Api-Key"] = input("Enter in the NR Accounts API key: ")
+headers["Content-Type"] = "application/json"
+
+# Opens and converts the New Relic json config into a python dictionary
+print("\nIf you are using another json file, please enter in the name. Otherwise leave the field blank to use the default json file.\n")
+JSON_FILE = input("JSON file name: ")
+if JSON_FILE:
+	DATA_FILE = open(JSON_FILE,)
+else:
+	DATA_FILE = open('NR-alert-config.json',)
+DATA = json.load(DATA_FILE)
+
+# Function to test if the API commands were successful and print the error message if they weren't
```

Comment:
FYI there is a built in function for checking request status codes..
https://requests.readthedocs.io/en/latest/_modules/requests/models/#Response.raise_for_status